nilearn/plotting/img_plotting.py

```diff
@@ -556,6 +559,163 @@ def plot_roi(roi_img, bg_img=MNI152TEMPLATE, cut_coords=None,
+            savefig.
+        cmap: matplotlib colormap, optional
+            The colormap for the atlas maps
+        vmin: float
+            Lower bound for plotting, passed to matplotlib.pyplot.imshow
+        vmax: float
+            Upper bound for plotting, passed to matplotlib.pyplot.imshow
+        alpha: float between 0 and 1
+            Alpha value sets the transparency of the color fillings inside the
+            contours.
+    """
+    display = plot_anat(anat_img, cut_coords=cut_coords,
+                        display_mode=display_mode,
+                        figure=figure, axes=axes, title=title,
+                        annotate=annotate, draw_cross=draw_cross,
+                        black_bg=black_bg, **kwargs)
+
+    maps_img = _utils.check_niimg_4d(maps_img)
+    n_maps = maps_img.shape[3]
+
+    _view_type = set(('auto', 'contours', 'filled_contours', 'continuous'))
+    if isinstance(view_type, _basestring):
+        if view_type not in _view_type:
+            raise TypeError('view_type option should be given '
+                            'either as a "contours" or "filled_contours" '
+                            'or "continuous" ')
+    else:
+        raise TypeError('view_type option should be '
+                        'either as a "contours" or "filled_contours" '
+                        'or "continuous" but you have given %s '
+                        % type(view_type))
+
+    cmap = plt.cm.get_cmap(cmap)
+    # Build a custom colormap for displaying contours
+    color_list = cmap(np.linspace(0, 1, n_maps))
+
+    if threshold is None:
+        # it will use default percentage,
+        # for a nicer look avoiding maximum overlaps for visualization
+        threshold = "99.7%"
+
+    if isinstance(threshold, list):
+        if len(threshold) != n_maps:
+            raise TypeError('The list of values to threshold '
+                            'should be equal to number of maps')
+    else:
+        threshold = [threshold] * n_maps
+
+    if view_type == 'auto':
+        if n_maps > 4:
+            view_type = 'contours'
+        else:
+            view_type = 'continuous'
+
+    for i, (map_img, color, thr) in enumerate(zip(iter_img(maps_img),
+                                                  color_list, threshold)):
+        data = map_img.get_data()
+        # To threshold or choose the level of the contours
+        thr = check_threshold(thr, data,
+                percentile_calculate=fast_abs_percentile, name='threshold')
+        # Get rid of background values in all cases
+        thr = max(thr, 1e-6)
+        if view_type.endswith('contours'):
+            display.add_contours(map_img, levels=[thr],
+                                 linewidths=linewidths,
+                                 colors=[color])
+            if view_type.startswith('filled'):
+                # Append the lower boundary value as 0 for contour fillings
+                display.add_contours(map_img, levels=[thr, 0.],
```

Comment:
Yes I tried with this idea but the only way to happen is to call both ```contour``` and ```contourf``` some where whether it would be in ```draw_2d``` or in the same ```plot_prob_atlas```. It looks complicated for me. Also we need to take care of ```levels``` which needs low level to be appended to ```0.``` to work for contour fillings seems to be bit complicated for me to handle both cases.